core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -10720,4 +10720,52 @@ private void checkLoptOptimizeJoinRule(LoptOptimizeJoinRule rule) {
     sql(sql).withRule(CoreRules.AGGREGATE_MIN_MAX_TO_LIMIT)
         .check();
   }
+
+  @Test void testOuterJoinForDphyp() {
```

Comment:
I am sorry to put forward my own opinion. I admire your algorithm transplantation ability and this algorithm may be very good, but referring to the standards of other computing engines, we may also need to introduce benchmarks or stress testing instances to determine how much performance improvement is obtained in what scenarios，the test cases do not include relevant benchmarks, so strictly speaking, it may not be possible to confirm that performance can be improved, so as to explain the benefits of this modification. @silundong